src/views/templates/actions/VirtualMachineTemplatesActions.tsx

```diff
@@ -0,0 +1,58 @@
+import * as React from 'react';
+
+import { V1Template } from '@kubevirt-ui/kubevirt-api/console';
+import { useKubevirtTranslation } from '@kubevirt-utils/hooks/useKubevirtTranslation';
+import { Action } from '@openshift-console/dynamic-plugin-sdk';
+import { Dropdown, DropdownItem, DropdownPosition, KebabToggle } from '@patternfly/react-core';
+
+import useVirtualMachineTemplatesActions from './hooks/useVirtualMachineTemplatesActions';
+
+type VirtualMachineTemplatesActionsProps = { template: V1Template };
+
+const VirtualMachineTemplatesActions: React.FC<VirtualMachineTemplatesActionsProps> = ({
+  template,
+}) => {
+  const { t } = useKubevirtTranslation();
+  const actions = useVirtualMachineTemplatesActions(template);
+  const [isOpen, setIsOpen] = React.useState(false);
+
+  const handleClick = (action: Action) => {
+    if (typeof action?.cta === 'function') {
+      action?.cta();
+      setIsOpen(false);
+    }
+  };
+
+  const disabledActionInfo = (actionId: string): string => {
+    switch (actionId) {
+      case 'edit-boot-source':
+        return t('Boot source cannot be edited for Red Hat templates');
+      case 'delete-template':
+        return t('Red Hat templates cannot be deleted');
+      default:
+        return t('This action is currently not avaliable');
+    }
+  };
+
+  return (
+    // TODO: use LazyActionMenu when fixed
+    <Dropdown
+      isPlain
+      isOpen={isOpen}
+      position={DropdownPosition.right}
+      toggle={<KebabToggle onToggle={setIsOpen} />}
+      dropdownItems={actions?.map((action) => (
+        <DropdownItem
+          key={action?.id}
+          onClick={() => handleClick(action)}
+          isDisabled={action?.disabled}
+        >
+          {action?.label}
+          {action?.disabled && <p>{disabledActionInfo(action?.id)}</p>}
```

Comment:
- action label does not change, the help text that help user know when this action is available and when not, is always displayed
- for the help under the action you can should use a description ( https://www.patternfly.org/v4/components/dropdown#dropdown-with-descriptions ) not primary action text